saleor/webhook/serializers.py

```diff
@@ -78,10 +78,11 @@ def serialize_checkout_lines_for_tax_calculation(
     discounts: Optional[Iterable[DiscountInfo]] = None,
 ) -> List[dict]:
     channel = checkout_info.channel
-
+    charge_taxes = get_charge_taxes_for_checkout(checkout_info, lines)
     return [
         {
             **_get_checkout_line_payload_data(line_info),
+            "charge_taxes": charge_taxes,
```

Comment:
We lose the whole feature here. It could be a breaking change if someone creates a tax app based on options to turn off taxes for some products. After upgrading to the new version we lose options to turn off taxes on one product without implementing tax_class in the tax app.

Reply:
In the new approach apps should use the subscription query to fetch the tax class assigned to a product (or product type) and then decide how to handle tax calculation based on it. I agree this might be breaking if anyone had an app that in 3.7.

Reply:
@fowczarek Can you describe a scenario that is impossible to solve using the new logic? Under the old logic, turning off "charge taxes" for a product was functionally the same as making the tax 0%.

Reply:
I don't know any scenario as an example. I think about it as a theoretical edge case.  We could change this behavior and update [docs here](https://docs.saleor.io/docs/3.x/developer/extending/apps/synchronous-webhooks/tax-webhooks#fields). We suggest creating a tax app with a usage subscription base webhooks. In the subscription base webhook, the developer is able to fetch any required data to calculate taxes including tax classes that are introduced in this feature.